Sources/protoc-gen-swift/StringUtils.swift

```diff
@@ -27,7 +31,12 @@ func splitPath(pathname: String) -> (dir:String, base:String, suffix:String) {
       suffix += String(c)
     }
   }
-  if suffix.characters.first != "." {
+#if swift(>=4.0)
+  let validSuffix = suffix.isEmpty || suffix.first == "."
```

Comment:
Is this the same behavior as before since it didn't check for empty before?

Reply:
If `suffix` is empty it just appended an empty string to `base` before, so it would not make a difference.